AwsCloudIdProvider.cs

```diff
@@ -114,6 +114,22 @@ public async Task<string> GetCloudIdAsync()
                 return SignRequest(accessKey, secretKey, securityToken);
             }
 
+            // if failed, try to take from environment
+            // Apperantly lambda functions use env vars and not profiles
+             try {
+                var profileCreds = new Amazon.Runtime.EnvironmentVariablesAWSCredentials();
+                var creds = await profileCreds.GetCredentialsAsync();
```

Comment:
this section is similar to the one above and below and can be unified to a function, input can be the type or the function to extract creds

Reply:
I don't think its worth it, since the initialization of the getter itself can throw. 
It will only "save" a few `x = y` lines.